src/vscode-dts/vscode.proposed.chatSessionsProvider.d.ts

```diff
@@ -5,30 +5,25 @@
 
 declare module 'vscode' {
 	/**
-	 * Provides a list of chat sessions
+	 * Provides a list of information about chat sessions.
 	 */
-	export interface ChatSessionsProvider extends Disposable {
+	export interface ChatSessionItemProvider {
```

Comment:
For api, providers aren't disposable in the API . Instead the extension could dispose of them when unregistering the provider if needed

Also we try to pass static information like `chatSesionType` in to the `register*` call